src/plugins/unified_doc_viewer/public/components/doc_viewer/doc_viewer.tsx

```diff
@@ -6,12 +6,22 @@
  * Side Public License, v 1.
  */
 
-import React from 'react';
+import React, { useMemo } from 'react';
 import type { DocViewRenderProps } from '@kbn/unified-doc-viewer/types';
 import { DocViewer } from '@kbn/unified-doc-viewer';
 import { getUnifiedDocViewerServices } from '../../plugin';
 
-export function UnifiedDocViewer(props: DocViewRenderProps) {
+export function UnifiedDocViewer({ docViewsRegistry, ...props }: DocViewRenderProps) {
```

Comment:
Hmm, I'm wondering if we can simplify this a little by removing the ability to pass a function here. Here's a patch showing kind of what I was thinking. Let me know if this works for you or if something is less ideal with that set of changes.

[patch.txt](https://github.com/elastic/kibana/files/13551479/patch.txt)

Reply:
I see your point here, there are a couple of reasons why I opted for this approach:
- Providing a function with a cloned registry removes the responsibility from each consumer to create a copy and avoid mutating the base registry, which helps to prevent unexpected doc views from appearing on other consumers, as each customization call will have its own registry.
- Passing only the registry implies adding the unified doc viewer plugin as a direct dependency on each plugin consuming it. This dependency is implicitly inherited when using Discover, so I thought it might be an unnecessary overhead.
- Using this signature users will still have the chance to apply the same pattern you proposed in the patch file, but also keep the chance to consume a registry instance using the function, forgetting about cloning responsibility.

Let me know your thoughts and any pros/cons with the 2 different approaches. Generally, I prefer to provide a `value | updater` signature for higher flexibility.